onnx/checker.cc

```diff
@@ -717,13 +714,24 @@ void check_opset_compatibility(
     return;
   }
 
-  const auto* schema_for_function_import =
-      ctx.get_schema_registry()->GetSchema(node.op_type(), func_opset_version, node.domain());
-
   const auto* schema_for_model_import =
       ctx.get_schema_registry()->GetSchema(node.op_type(), model_opset_version, node.domain());
 
-  if (schema_for_function_import && schema_for_model_import &&
+  // If func_opset_version > model_opset_version then it is possible that the opschema
+  // for this node is not available. This is acceptable as long as schema is present for the
+  // version imported by function opset imports.
+  // Example: If customdomain.Foo is converted (in place .i.e without version bump) from
+  // primitive op to function op containing nodes from onnx domain then an older model which
+  // contains Foo can run into this scenario. This is bcecause Foo will import a later
+  // version of onnx domain than this older model.
```

Comment:
I am not sure about this. I think we can allow func_opset_version > model_opset_version as long as both schemas and exist and we can check the equality of since_version for both as below.